File: python/resistor-color-duo/resistor_color_duo.py
COLORS = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white",
]


def value(colors):
    return (resistenceValueOf(colors[0]) * 10) + resistenceValueOf(colors[1])


def resistenceValueOf(color):
    return COLORS.index(color)

File: typescript/resistor-color-duo/resistor-color-duo.ts
const COLOR_TO_VALUE = {
  black: 0,
  brown: 1,
  red: 2,
  orange: 3,
  yellow: 4,
  green: 5,
  blue: 6,
  violet: 7,
  grey: 8,
  white: 9,
};

type Color = keyof typeof COLOR_TO_VALUE;

export class ResistorColor {
  private colors: Color[];

  constructor(colors: Color[]) {
    if (colors.length < 2) {
      throw new Error("At least two colors need to be present");
    }

    this.colors = colors;
  }

  value() {
    const [firstColor, secondColor] = this.colors;

    return COLOR_TO_VALUE[firstColor] * 10 + COLOR_TO_VALUE[secondColor];
  }
}
